exercises/stereo/descr.md

```diff
@@ -0,0 +1,132 @@
+# Trees in Stereo Vision
+
+This short exercise considers a simple yet challenging question: *can one
+reconstruct a binary tree out of its inorder and preorder traversals?* This
+problem was introduced by Knuth in "The Art of Computer Programming" (1968).
+
+## Tree Traversals
+
+Here is an algebraic data type of binary trees:
+
+```
+  type 'a tree =
+    | Leaf
+    | Node of 'a tree * 'a * 'a tree
+```
+
+A tree `t` is either a leaf `Leaf` or a binary node `Node (u, x, v)`,
+which carries a left subtree `u`, an element `x`, and a right subtree `v`.
+
+A *traversal* of a tree `t` is a list of the elements carried by the tree `t`.
+There are several natural ways of ordering such a list. At a binary node `Node
+(u, x, v)`, three objects are at hand, namely `u`, `x`, `v`. There are six
+permutations of these objects. These six permutations give rise to six natural
+traversals, which are known as *inorder*, *preorder*, *postorder*, *reverse
+inorder*, *reverse preorder*, and *reverse postorder* traversals. In this
+exercise, two of these are of interest:
+
+* The *inorder traversal* corresponds to the permutation `u`, `x`, `v`. It is
```

Comment:
It may not be clear for the reader which permutation is encoded by the sequence: "`u`, `x`, `v`. Here you could say:
```suggestion
* The *inorder traversal* corresponds to the identity permutation. It is
```